pkg/logql/quantile_over_time_sketch.go

```diff
@@ -231,18 +231,36 @@ func (r *quantileSketchBatchRangeVectorIterator) agg(samples []promql.FPoint) sk
 	return s
 }
 
+// Initializing pool
+var quantileVectorPool = sync.Pool{
+	// New optionally specifies a function to generate
+	// a value when Get would otherwise return nil.
+	// New: func() interface{} { return make([]ProbabilisticQuantileVector, 2) },
+	New: func() interface{} { return make([]ProbabilisticQuantileVector, 0, 2) },
```

Comment:
@cstyan why the length of two?

Reply:
if we're doing a merge we have at least 2 sketches